packages/types/lib/driver.ts

```diff
@@ -101,20 +102,60 @@ export interface FindCommands {
   getPageSource(): Promise<string>;
 }
 
-export interface LogCommands {
-  supportedLogTypes: Record<string, LogType<Driver>>;
-  getLogTypes(): Promise<string[]>;
+/**
+ * Log-related functionality of a {@linkcode Driver}. To be used as a mixin
+ */
+export interface ILogCommands<C extends Constraints> {
   /**
-   * Gets logs
+   * Definition of the available log types
+   */
+  supportedLogTypes: Readonly<LogDefRecord<C>>;
+
+  /**
+   * Get available log types as a list of strings
+   */
+  getLogTypes(): Promise<(keyof ILogCommands<C>['supportedLogTypes'])[]>;
```

Comment:
gotta love how we reimplement the function logic in types

Reply:
yeah, there's a limit to its type inference.  Unless we told it otherwise, it'd think the return type was just `string[]`, even if all members are `keyof SomeType`.